src/link/link.service.ts

```diff
@@ -0,0 +1,70 @@
+import { Injectable, HttpException, HttpStatus } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { User } from '../entities/user.entity';
+import { ElderlyCode, ElderlyProfile, CaretakerInfo } from './interfaces/link.interface'
+import Hashids from 'hashids';
+
+@Injectable()
+export class LinkService {
+    constructor(
+        @InjectRepository(User)
+        private userRepository: Repository<User>,
+    ) {}
+
+    async getElderly(elderlyCode: string): Promise<ElderlyProfile>{
+        const hashids = new Hashids('aegis', 6, 'ABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890');
```

Comment:
this and line 59 is the same function and it use the same key to encode and decode, therefore we should keep the key in the same place(and then when we want to change or do something with key, we just change it in one place and I think it should be keep in env file(and post in secret channel in discord too)